realsense/plugin.cpp

```diff
@@ -110,43 +182,72 @@ class realsense : public plugin {
+                            img1 = cam_D4XX_.img1;
+                            rgb = cam_D4XX_.rgb;
+                            depth = cam_D4XX_.depth;
+                        }
+                        
+                        // Submit to switchboard
+                        _m_imu_cam->put(new imu_cam_type{
+                                imu_time_point,
+                                av,
+                                la,
+                                img0,
+                                img1,
+                                imu_time,
+                            });
+                        
+                        if (rgb && depth)
+                        {
+                            _m_rgb_depth->put(new rgb_depth_type{
+                                    rgb,
+                                    depth,
+                                    imu_time,
+                                });
+                        }
+            
+                        auto imu_integrator_params = new imu_integrator_seq{
+                            .seq = static_cast<int>(++_imu_integrator_seq),
+                        };
+                        _m_imu_integrator->put(imu_integrator_params);
                     }
 
-                    // Submit to switchboard
-                    _m_imu_cam->put(new imu_cam_type{
-                            imu_time_point,
-                            av,
-                            la,
-                            img0,
-                            img1,
-                            imu_time,
-                        });
-
-                    if (rgb && depth)
-                    {
-                        _m_rgb_depth->put(new rgb_depth_type{
-                                rgb,
-                                depth,
+                    else if (realsense_cam.compare("T26X") == 0){
+                        std::optional<cv::Mat *> img0 = std::nullopt;
+                        std::optional<cv::Mat *> img1 = std::nullopt;
+
+                        if (last_iteration_cam != cam_T26X_.iteration)
+                        {
+                            last_iteration_cam = cam_T26X_.iteration;
+                            img0 = cam_T26X_.img0;
+                            img1 = cam_T26X_.img1;
+                        }
+                
+                        // Submit to switchboard
+                        _m_imu_cam->put(new imu_cam_type{
+                                imu_time_point,
+                                av,
+                                la,
+                                img0,
+                                img1,
                                 imu_time,
                             });
+            
+                        auto imu_integrator_params = new imu_integrator_seq{
+                            .seq = static_cast<int>(++_imu_integrator_seq),
+                        };
+                        _m_imu_integrator->put(imu_integrator_params);
```

Comment:
Another heads up: The `imu_integrator_seq` topic is removed from the plugin in [#32](https://github.com/ILLIXR/ILLIXR/blob/865edb7cfe2e13dce2f0972cd3840c61889afeae/realsense/plugin.cpp#L154).